### PR TITLE
Squashed/rebased #1373

### PR DESCRIPTION
**Pull request by twolfson**
As requested by @jzaefferer in #1512, we have taken care of squashing/rebasing #1373 while maintaining appropriate authorship. To make this easier for future persons, these are the steps I took:

```bash
# Added upstream remote
git remote add upstream https://github.com/jzaefferer/jquery-validation

# Installed `upstream-pr` branch
git config --add remote.upstream-pr.url $(git config --get remote.upstream.url) && git config --add remote.upstream-pr.fetch +refs/pull/*/head:refs/remotes/upstream/pr/*

# Fetched upstream PRs
git fetch upstream-pr

# Checked out PR in question
git checkout pr/1373

# Checked out branch for squashing
git checkout pr/1373.squash

# Squashed commits into 1 commit
# Use `git log` to find commit where we diverged from `master
git rebase -i {{last commit from master}}
# Select all but head commit as "squash"

# Check out new branch to perform rebasing
git checkout pr/1373.rebased

# Rebase 1 commit on top of `upstream/master`
git rebase -i upstream/master

# Handle merge conflicts
git mergetool -y

# Verify that diff aligns with current GitHub PR
# Should see no changes here
git diff pr/1373.squash -- src/additional/accept.js
# Should align with diff on GitHub
git diff upstream/master

# Complete our rebase
git rebase --continue

# All done, PR opened
```

/cc @jzaefferer

### CODE REVIEW

**Comment by Arkni** [comment]
@staabm 
This should be merged by now.

**Comment by staabm** [comment]
@Arkni missed this PR... could you rebase it against master?

**Comment by twolfson** [comment]
This PR is a rebase of #1373 and this was opened 4 months ago under the presumption that it would be landed more/less in a timely manner (i.e. without merge conflict). What's the guarantee that this second rebase will be landed in a timely manner?

**Comment by staabm** [comment]
@twolfson I willing to land it and now also have the required permissions todo it :-)

**Comment by twolfson** [comment]
Alright, taking a look at rebasing now

**Comment by Arkni** [comment]
The test is failing due to this check ([src/core.js#L643](https://github.com/jzaefferer/jquery-validation/blob/master/src/core.js#L643-L647)):
```js
if ( element.hasAttribute( "contenteditable" ) ) {
	val = $element.text();
} else {
	val = $element.val();
}
```

The mocked input file does not have the |hasAttribute` method. A quick fix could be:
```diff
 test/methods.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/methods.js b/test/methods.js
index 0848357..cf29f04 100644
--- a/test/methods.js
+++ b/test/methods.js
@@ -37,7 +37,8 @@ function acceptFileDummyInput( filename, mimeType ) {
                type: "file",
                files: fileList,
                nodeName: "INPUT",
-               value: "/tmp/fake_value"
+               value: "/tmp/fake_value",
+               hasAttribute: function() { return false; }
        };
 }
```

**Comment by twolfson** [comment]
Sorry, got side-tracked on other work. Thanks for the pointer. I will try that out in a bit

**Comment by twolfson** [comment]
@Arkni That patch worked great, thanks =)

@staabm This PR should be good to go

File: test/methods.js [hunk 1]
```diff
@@ -1309,3 +1342,24 @@ test( "cpfBR", function() {
 	ok( !method( "11144477715" ), "Invalid CPF Number: 1st check number failed" );
 	ok( !method( "11144477737" ), "Invalid CPF Number: 2nd check number failed" );
 } );
+
+test( "file accept - image wildcard", function() {
+	var input = acceptFileDummyInput( "test.png", "image/png" ),
+			$form = $( "<form />" ),
+			proxy = $.proxy( $.validator.methods.accept, new $.validator( {}, $form[ 0 ] ), null, input, "image/*" );
```
Comment (staabm) [inline test/methods.js]: Where does this `$.proxy` come from?

Comment (twolfson) [inline test/methods.js]: That was in the initial PR (#1373) but it's defined via jQuery and essential is ES5's `Function.bind`:

https://api.jquery.com/jQuery.proxy/

**Comment by staabm** [comment]
Will land it tomorrow after beeing back on the oc. Thx so far.

**Comment by twolfson** [comment]
Sweet, thanks =)

**Comment by staabm** [comment]
@twolfson landed it right now after a few minor changes in the commit message.

thank you!

**Comment by twolfson** [comment]
Cool, thanks =)

**Pull request closed without merging**